components/api/filter/validator-filter.bal

```diff
@@ -0,0 +1,62 @@
+import ballerina/config;
+import ballerina/http;
+import ballerina/log;
+import ballerina/time;
+import cellery_hub_api/validator;
+import cellery_hub_api/constants;
+import ballerina/mysql;
+import ballerina/sql;
+
+public type validateRequestFilter object {
+    public function filterRequest(http:Caller caller, http:Request request,
+                        http:FilterContext context)
+                        returns boolean {
+        log:printDebug("Request was intercepted to validate the validate the token ......");
+        string token = "";
+        if(request.hasHeader(constants:AUTHERIZATION_TOKEN)) {
+            token = request.getHeader(constants:AUTHERIZATION_TOKEN);
+            if "" == token {
+                log:printInfo("Did not receive any token. 
+                Passing the request to the next filter");
+                request.removeHeader(constants:AUTHENTICATED_USER);
+                return true;
+            }
+            validator:Conf|error conf = untaint validator:getAuthConfig();
+            if conf is validator:Conf {
+                string|error username = validator:getUsername(untaint token, conf);
+                if (username is string) {
+                    if username != "" {
+                        request.setHeader(constants:AUTHENTICATED_USER, username);
+                        log:printInfo("The token is valid");
```

Comment:
Logs must be comprehensive and self contained. For an example when concurrent requests are going through, the log "Token is valid" doesn't make sense. You can add more information like username and other information if available.

Reply:
Don't we need to check the username with the received username to check whether they are matching ?

Reply:
I am not checking the incoming username via header. I am replacing the username with the username contain in the token. If there is no username in the token I am removing the header.